xbmc/music/tags/MusicInfoTag.cpp

```diff
@@ -867,6 +893,22 @@ void CMusicInfoTag::Archive(CArchive& ar)
     ar >> m_lastPlayed;
     ar >> m_dateAdded;
     ar >> m_strComment;
+    int iMusicRolesSize;
+    ar >> iMusicRolesSize;
+    m_musicRoles.reserve(iMusicRolesSize);
+    for (int i = 0; i < iMusicRolesSize; ++i)
+    {
+      int idRole;
+      long idArtist;
+      std::string strArtist;
+      std::string strRole;
+      ar >> idRole;
+      ar >> strRole;
+      ar >> strArtist;
+      ar >> idArtist;
+      CMusicRole Credit(idRole, strRole, strArtist, idArtist);
+      m_musicRoles.push_back(Credit);
```

Comment:
can be called as m_musicRoles.emplace_back(idRole, strRole, strArtist, idArtist); to avoid copying the item.